overlord/snapstate/snapstate.go

```diff
@@ -820,7 +820,20 @@ func RestartSystem(task *state.Task, rebootInfo *boot.RebootInfo) {
 	if immediate {
 		rst = restart.RestartSystemNow
 	}
-	restart.Request(task.State(), rst, rebootInfo)
+
+	return FinishTaskWithRestart(task, status, rst, rebootInfo)
+}
+
+// FinishTaskWithRestart will finish any task that needs a restart.
+// TODO for the moment it just calls restart.Request, but we will
+// return an error in the future if we want to hold the restart. We
+// will also add post hold status and boot id information in the task
+// in that case.
+func FinishTaskWithRestart(task *state.Task, status state.Status, rt restart.RestartType,
+	rebootInfo *boot.RebootInfo) error {
+
```

Comment:
as we discussed this should set status on task

Reply:
Done